### PR TITLE
added links to list of available JVMs

### LINKED ISSUE #1482: Add link to up-to-date list of supported JDKs to https://docs.travis-ci.com/user/languages/java/#Testing-Against-Multiple-JDKs
The [Testing against multiple JDKs section of the docs](https://docs.travis-ci.com/user/languages/java/#Testing-Against-Multiple-JDKs) mentions

    - oraclejdk8
    - oraclejdk9
    - openjdk7

as an example of testing against multiple JDKs, however it's not clear whether this is the complete list or not.

### PR DESCRIPTION
**Pull request by krichter722**
added links to list of available JVMs to "Testing Against multiple JDKs" in java.md, closes #1482.

Please note that this might contradict the idea to keep OS-specific reference out of test because there's a note saying that OS-specific stuff is on a separate page as mentioned in #1482, but why not just drop that and put the links where they're expected and thus most useful.

### CODE REVIEW

**Comment by plaindocs** [comment]
I'm ok with this approach. 

In fact we should probably remove the generic links from the note, and substitute language specific links on every page. We *should* use markdown link syntax instead of displaying the full URL though.

File: user/languages/java.md [hunk 1]
```diff
@@ -185,6 +185,12 @@ jdk:
 the [OS X Build Environment](/user/reference/osx/#JDK-and-OS-X) for more
 details.
 
+The list of available JVMs for different dists are at
+
+  * for Trusty: https://docs.travis-ci.com/user/reference/trusty/#JVM-(Clojure%2C-Groovy%2C-Java%2C-Scala)-images
+  * for Precise: https://docs.travis-ci.com/user/reference/precise/#JVM-(Clojure%2C-Groovy%2C-Java%2C-Scala)-VM-images
```
Comment (olleolleolle) [inline user/languages/java.md]: Maybe something like:

```

* [JDKs installed for **Trusty**](https://docs.travis-ci.com/user/reference/trusty/#JVM-(Clojure%2C-Groovy%2C-Java%2C-Scala)-images)
* [JDKs installed for **Precise**](https://docs.travis-ci.com/user/reference/precise/#JVM-(Clojure%2C-Groovy%2C-Java%2C-Scala)-VM-images)

```

**Comment by olleolleolle** [comment]
@plaindocs This patch saved my day today.

**Comment by plaindocs** [comment]
Thanks @krichter722

**Pull request merged**